### PR TITLE
New tests

### PR DESCRIPTION
**Pull request by cool-user**

### CODE REVIEW

File: packages/hw-app-ada/test/src/current/direct/getExtendedPublicKey.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      );
+
+      throw new Error(codeTooFarMessage);
+    } catch (error) {
+      expect(error.message).not.to.have.string(codeTooFarMessage);
+    }
+
+    try {
+      // invalid P2
+      await transport.send(
```
Comment (ppershing) [inline packages/hw-app-ada/test/src/current/direct/getExtendedPublicKey.js]: This seems too repetitive, can we move it to some lambda?

File: packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  beforeEach(async () => {
+    ada = await getAda();
+  });
+
+  afterEach(async () => {
+    await ada.t.close();
+  });
+
+  it("Should successfully get extended public key", async () => {
```
Comment (ppershing) [inline packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js:19]: Does this work as expected?
In jest we had to declare upfront the number of assertions in the async case in orther for the checking to work but maybe Mocha is different ...

File: packages/hw-app-ada/test/src/pathDerivations.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+export default {
```
Comment (ppershing) [inline packages/hw-app-ada/test/src/pathDerivations.js]: +1, nice idea

File: packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    const der1 = pathDerivations["44'/1815'/1'"];
+    const der2 = pathDerivations["44'/1815'/1'/0/12'"];
+    const der3 = pathDerivations["44'/1815'/1'/0/10'/1/2/3"];
+
+    const result1 = await ada.getExtendedPublicKey(pathToArray(der1.path));
```
Comment (ppershing) [inline packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js]: This whole function seems to be too repetitive ...
Why not extract common logic to a lambda?

File: packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js (hunk 2 above)
Comment (cool-user) [reply packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js:19]: Looks to be working properly

File: packages/hw-app-ada/src/Ada.js [hunk 5]
```diff
@@ -19,13 +19,13 @@
...
 
-const CLA = 0xD7;
+const CLA = 0xd7;
 const HARDENED = 0x80000000;
 
 const INS_GET_VERSION = 0x00;
 const INS_GET_EXT_PUBLIC_KEY = 0x10;
 const INS_ATTEST_UTXO = 0x20;
-const INS_RUN_TESTS = 0xF0;
+const INS_RUN_TESTS = 0xf0;
```
Comment (ppershing) [inline packages/hw-app-ada/src/Ada.js:28]: Hmm, I am wondering whether we shouldn't replace this with
const INS = {GET_VERSION = 0x00, ...} enum object. No needed in this PR though

Comment (cool-user) [reply packages/hw-app-ada/src/Ada.js:28]: yes, might do in next PR

File: packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
+import { expect } from "chai";
+import pathDerivations from "../../pathDerivations";
+
+import { getAda, pathToArray } from "../../utils";
+
+describe("getExtendedPublicKey", async () => {
+  let ada = {};
+
+  beforeEach(async () => {
+    ada = await getAda();
```
Comment (ppershing) [inline packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js:12]: `getAda` seems to be quite a bad name

File: packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    } catch (error) {
+      expect(error.message).to.have.string("5003");
+    }
+  });
+
+  it("Should reject path not starting with 44'/1815'/n'", async () => {
+    try {
+      await ada.getExtendedPublicKey(pathToArray("44'/1815'/33/125"));
+
+      throw new Error("Expected error");
```
Comment (ppershing) [inline packages/hw-app-ada/test/src/current/integration/getExtendedPublicKey.js]: I don't think this is "expected error", rather it is "should have thrown before this line"

File: packages/hw-app-ada/test/src/pathDerivations.js (hunk 3 above)
Comment (ppershing) [inline packages/hw-app-ada/test/src/pathDerivations.js]: maybe place this under __fixtures__ directory?

**Review by ppershing — APPROVED** [review]

**Pull request merged**